glass/_array_api_utils.py

```diff
@@ -42,78 +111,202 @@ def get_namespace(*arrays: AnyArray) -> ModuleType:
 
 
 def rng_dispatcher(
-    array: NDArray[Any] | JAXArray,
+    array: AnyArray,
 ) -> np.random.Generator | glass.jax.Generator | Generator:
-    """Dispatch RNG on the basis of the provided array."""
-    backend = array.__array_namespace__().__name__
+    """
+    Dispatch a random number generator based on the provided array's backend.
+
+    Parameters
+    ----------
+    array : AnyArray
+        The array whose backend determines the RNG.
+
+    Returns
+    -------
+    np.random.Generator | glass.jax.Generator | Generator
+        The appropriate random number generator for the array's backend.
+
+    Raises
+    ------
+    NotImplementedError
+        If the array backend is not supported.
+    """
+    xp = get_namespace(array)
+    backend = xp.__name__
+
     if backend == "jax.numpy":
         import glass.jax  # noqa: PLC0415
 
         return glass.jax.Generator(seed=42)
+
     if backend == "numpy":
-        return np.random.default_rng()
+        return xp.random.default_rng()  # type: ignore[no-any-return]
```

Comment:
This issue with mypy seems to be something we have to put up with. I wonder if through `ty` whether it can be handled in a better way.